databricks/koalas/frame.py

```diff
@@ -4372,6 +4372,9 @@ def drop(self, labels=None, axis=1,
                               in zip(self._internal.data_columns, self._internal.column_index)
                               if idx not in drop_column_index))
             internal = self._internal.copy(data_columns=list(cols), column_index=list(idx))
+            # To drop internal Spark DataFrame's columns also
+            for col in columns:
+                internal._sdf = internal._sdf.drop(col[0])
```

Comment:
Each call of `drop` creates Spark DataFrame which accumulates a bit of overhead.

Reply:
ditto !